test/lib/commandFactory.test.js

```diff
@@ -0,0 +1,312 @@
+                });
+            });
+        });
+
+        it('creates a Command and auto-bumps version when latest returns something', () => {
+            const latest = {
+                namespace,
+                command,
+                version: `${version}.0`,
+                maintainer,
+                description,
+                format,
+                habitat,
+                id: generatedId
+            };
+
+            expected.version = `${version}.1`;
+
+            datastore.save.resolves(expected);
+            datastore.scan.resolves([latest]);
+
+            return factory.create({
+                namespace,
+                command,
+                version,
+                maintainer,
+                description,
+                format,
+                habitat
+            }).then((model) => {
+                assert.instanceOf(model, Command);
+                Object.keys(expected).forEach((key) => {
+                    assert.strictEqual(model[key], expected[key]);
+                });
+            });
+        });
+    });
+
+    describe('getInstance', () => {
+        let config;
+
+        beforeEach(() => {
+            config = { datastore };
+        });
+
+        it('should get an instance', () => {
+            const f1 = CommandFactory.getInstance(config);
+            const f2 = CommandFactory.getInstance(config);
+
+            assert.instanceOf(f1, CommandFactory);
+            assert.instanceOf(f2, CommandFactory);
+
+            assert.equal(f1, f2);
+        });
+
+        it('should throw when config not supplied', () => {
+            assert.throw(CommandFactory.getInstance,
+                Error, 'No datastore provided to CommandFactory');
+        });
+    });
+
+    describe('getCommand', () => {
+        const commandNamespace = 'testCommandNS';
+        const commandName = 'testCommand';
+        const commandVersion = '1.0';
+        let fullCommandName;
+        let expected;
+        let returnValue;
+
+        beforeEach(() => {
+            fullCommandName = `${commandNamespace}/${commandName}@${commandVersion}`;
+
+            returnValue = [
+                {
+                    id: '1',
+                    namespace: 'testCommandNS',
+                    command: 'testCommand',
+                    version: '1.0.1'
+                },
+                {
+                    id: '3',
+                    namespace: 'testCommandNS',
+                    command: 'testCommand',
+                    version: '1.0.3'
+                },
+                {
+                    id: '2',
+                    namespace: 'testCommandNS',
+                    command: 'testCommand',
+                    version: '1.0.2'
+                },
+                {
+                    id: '4',
+                    namespace: 'testCommandNS',
+                    command: 'testCommand',
+                    version: '2.0.1'
+                }
+            ];
+        });
+
+        it(
```

Comment:
A new line is here. Do you need it?

Reply:
Yes, it is. If there is no new line, `eslint` check fails because of max line length violation.

Reply:
Understood. Thanks!